[feature] add seatunnel provider

ocean-zhc opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

ocean-zhc committed:
[feature] add seatunnel provider

ocean-zhc committed:
[feature] add seatunnel provider

ocean-zhc committed:
[feature] add seatunnel provider

ocean-zhc committed:
[feature] add seatunnel provider

Hisoka-X commented on providers/apache/seatunnel/tests/seatunnel_sensor_example_dag.py:
```diff
@@ -0,0 +1,120 @@
+    'retry_delay': timedelta(minutes=5),
+}
+
+# 定义用于从命令输出中提取job_id的函数
```
The chinese comment should be convert to english.

ocean-zhc replied:
+1

ocean-zhc committed:
Merge branch 'apache:main' into main

ocean-zhc committed:
Merge branch 'apache:main' into main

ocean-zhc committed:
[feature] add seatunnel integration tests

ocean-zhc committed:
Merge remote-tracking branch 'origin/main'

ocean-zhc committed:
Merge branch 'main' into main

ocean-zhc committed:
Merge branch 'apache:main' into main

ocean-zhc committed:
[feature] add seatunnel integration tests

ocean-zhc committed:
Merge branch 'main' into main

ocean-zhc committed:
Merge branch 'apache:main' into main

ocean-zhc committed:
[feature] add seatunnel integration tests

ocean-zhc committed:
[feature] add seatunnel integration tests

ocean-zhc committed:
Fix pre-commit issues for SeaTunnel provider

ocean-zhc committed:
Complete SeaTunnel provider fixes - all CI issues resolved

- Fixed all import paths for Airflow 3.0 SDK
- Added source-date-epoch to provider.yaml
- Updated boring-cyborg configuration
- Generated required provider build files
- Fixed all pre-commit hook issues
- All static checks, mypy, and formatting now pass

ocean-zhc committed:
Update minimum Airflow version requirement to 2.10.0

- Updated provider.yaml, pyproject.toml, and all documentation
- Updated test requirements and integration test docs
- Consistent version requirements across all files
- Required for Airflow 3.0 SDK compatibility

ocean-zhc committed:
Fix end-of-file newline issue in commits.rst

ocean-zhc committed:
Fix import issues in test files

- Changed from airflow.models import Connection to airflow.models.connection import Connection
- Fixed DAG import to use airflow.sdk import DAG
- These changes ensure compatibility with Airflow 3.0 SDK structure

ocean-zhc committed:
Merge branch 'main' into main
